website/notifications/emails.py

```diff
@@ -52,21 +52,22 @@ def notify(event, user, node, timestamp, **context):
     return sent_users
 
 def notify_mentions(event, user, node, timestamp, **context):
+    new_mentions = context.get('new_mentions', [])
+    sent_users = notify_global_event(event, user, node, timestamp, new_mentions, **context)
+    return sent_users
+
+def notify_global_event(event, sender_user, node, timestamp, target_users, **context):
     event_type = utils.find_subscription_type(event)
     sent_users = []
-    new_mentions = context.get('new_mentions', [])
-    for m in new_mentions:
-        mentioned_user = OSFUser.load(m)
-        subscriptions = get_user_subscriptions(mentioned_user, event_type)
+
+    for target_id in target_users:
+        target_user = OSFUser.load(target_id)
+        subscriptions = get_user_subscriptions(target_user, event_type)
         for notification_type in subscriptions:
-            if (
-                notification_type != 'none' and
-                subscriptions[notification_type] and
-                m in subscriptions[notification_type]
-            ):
-                store_emails([m], notification_type, 'mentions', user, node,
-                                 timestamp, **context)
-                sent_users.extend([m])
+            if (notification_type != 'none' and subscriptions[notification_type] and target_id in subscriptions[notification_type]):
+                store_emails([target_id], notification_type, event, sender_user, node, timestamp, **context)
```

Comment:
Minor (feel free to ignore): Since `store_emails` can take a list of recipients, it seems like it would be nicer to first collect the user subscriptions together, then call `store_emails` just once per `notification_type`